client/hostvolumemanager/host_volumes_test.go

```diff
@@ -21,10 +23,12 @@ func TestNewHostVolumeManager_restoreState(t *testing.T) {
 	vol := &cstructs.HostVolumeState{
 		ID: "test-vol-id",
 		CreateReq: &cstructs.ClientHostVolumeCreateRequest{
+			Name:     "test-vol-name",
 			ID:       "test-vol-id",
 			PluginID: "mkdir",
 		},
 	}
+	fNode := newFakeNode()
```

Comment:
If it's at all feasible, I'd considering using a real client in here rather than a mock. There's lock logic that'd be nice to make sure is getting exercised.

Reply:
I figure the actual client usage is being exercised in the client RPC endpoint test:
https://github.com/hashicorp/nomad/pull/24613/files#diff-054b6367d0da40a926f9565f6129bf0fabc62f7c2bb154540a8c69ac5f10b5cdR30-R31

I don't think this thing down here needs to know what a client is.  also, I think I'd hit an import cycle if I import `client`, since `client` imports this.

Reply:
Fair enough.